ttnn/cpp/ttnn/tensor/tensor.hpp

```diff
@@ -184,11 +181,12 @@ struct Tensor {
     // ======================================================================================
     const Storage &get_storage() const;
     // [[deprecated("Use get_shape() instead.")]]
-    const tt::tt_metal::LegacyShape &get_legacy_shape() const;
-    const ttnn::Shape &get_shape() const;
-    const DataType &get_dtype() const;
-    const Layout &get_layout() const;
-    const Tile &get_tile() const;
+    tt::tt_metal::LegacyShape get_legacy_shape() const;
+    ttnn::Shape get_shape() const;
+    DataType get_dtype() const;
+    Layout get_layout() const;
+    Tile get_tile() const;
```

Comment:
must be removed and accessed from TensorSpec or TensorLayout. worst case optional.